tools-local/unix/acquire-tool.sh

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env bash
+
+# Downloads the declared version of the specified tool.
+# Download URL and package name corresponding to the tool is read from the .toolversions file.
+# Arguments:
+#   1. Name of the tool
+
+if [ "$#" -ne 1 ]; then
+    echo "Usage: $0 ToolName"
+    echo "          ToolName: Name of the tool"
+fi
+
+if [ -z "$1" ]; then
+    echo "Argument passed as tool name is empty. Please provide a non-empty string."
+    exit 1
+fi
+
+toolName="$1"
+
+# Downloads the package corresponding to the tool, and extracts the package.
+download_extract()
+{
+    # Get the download URL
+    downloadUrl="$(get_download_url "$toolName")"
+
+    if [ $? -ne 0 ]; then
+        echo "$downloadUrl"
+        exit 1
+    fi
+
+    # Get the package name corresponding to the tool.
+    downloadPackageName=$(get_download_package_name "$toolName")
+
+    # Create folder to save the downloaded package, and extract the package contents.
+    repoTools=$(get_repository_tools_downloads_folder "$toolName")
+    toolFolder="$repoTools/$toolName"
+    rm -rf "$toolFolder"
+    mkdir -p "$toolFolder"
+
+    # Download
+    curl --retry 10 -ssl -v --output "$repoTools/$toolName/$downloadPackageName" "$downloadUrl$downloadPackageName" 2> "$repoTools/$toolName/download.log"
```

Comment:
Added a fallback to wget.

---
In reply to: [107990473](https://github.com/ravimeda/corefx/pull/2#discussion_r107990473) [](ancestors = 107990473)